src/Compilers/VisualBasic/Portable/Analysis/FlowAnalysis/ReadWriteWalker.vb

```diff
@@ -143,6 +144,21 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                     If Not (Me._currentMethodOrLambda = param.ContainingSymbol) Then
                         Me._captured.Add(param)
                     End If
+                Case SymbolKind.RangeVariable
+                    Dim range = DirectCast(variable, RangeVariableSymbol)
+                    If Me._currentMethodOrLambda = range.ContainingSymbol Then
+                        ' OK, defined in same method, so not captured
+                    ElseIf Me._currentQueryLambda Is Nothing Then
+                        ' Not in a query lambda, but defined and used in different methods, so captured.
+                        Me._captured.Add(range)
+                    Else
+                        ' If in a nested lambda, or
+                        ' Not one of the range variables of the current query, then captured
+                        If Not (Me._currentMethodOrLambda = Me._currentQueryLambda.LambdaSymbol) OrElse
+                            Not (Me._currentQueryLambda.RangeVariables.Contains(range)) Then
+                            Me._captured.Add(range)
```

Comment:
I mentioned this for C# change as well. It feels to me that, in order for this information to be useful, capture status of a range variable should be determined in relation to the region being analyzed.